pkg/workload/workload.go

```diff
@@ -1081,7 +1081,11 @@ func AdmissionChecksForWorkload(log logr.Logger, wl *kueue.Workload, admissionCh
 }
 
 func Evict(ctx context.Context, c client.Client, recorder record.EventRecorder, wl *kueue.Workload, reason, underlyingCause, msg string, clock clock.Clock) error {
-	prepareForEviction(wl, clock.Now(), reason, msg)
+	if reason == kueue.WorkloadDeactivated && underlyingCause != "" {
+		prepareForEviction(wl, clock.Now(), fmt.Sprintf("%sDueTo%s", reason, underlyingCause), msg)
+	} else {
+		prepareForEviction(wl, clock.Now(), reason, msg)
+	}
```

Comment:
```suggestion
	if reason == kueue.WorkloadDeactivated && underlyingCause != "" {
		reason = fmt.Sprintf("%sDueTo%s", reason, underlyingCause)
	}
	prepareForEviction(wl, clock.Now(), reason, msg)
```

The differences between those are only `reason`. So, the overriding reason makes more sense.